### PR TITLE
Fix #9986, implemented rax2 -i functionality, to mimic xxd -i

### LINKED ISSUE #9986: Implement xxd -i in rax2 -i
```
$ echo hello world > hw
$ xxd -i hw
unsigned char hw[] = {
  0x68, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x77, 0x6f, 0x72, 0x6c, 0x64, 0x0a
};
unsigned int hw_len = 12;
$
```

This is already implement in r2 via the pc command, we just need to expose it in rax2 and update the manpage

### PR DESCRIPTION
**Pull request by PaoloMonti42**
I would have loved to use r_print_code, but for some reason it's not visible from rax2.c

### CODE REVIEW

File: binr/rax2/rax2.c [hunk 1]
```diff
@@ -480,7 +482,26 @@ static int rax(char *str, int len, int last) {
 			free (str);
 		}
 		return false;
-	}
+	} else if (flags & (1 << 21)) { // -i
+                printf("unsigned char buf[] = {");
+                /* resonable amount of bytes per line */
+                int byte_per_col = 12;
```
Comment (radare) [inline binr/rax2/rax2.c]: use tabs

Comment (radare) [inline binr/rax2/rax2.c]: this is const

File: binr/rax2/rax2.c [hunk 2]
```diff
@@ -480,7 +482,26 @@ static int rax(char *str, int len, int last) {
 			free (str);
 		}
 		return false;
-	}
+	} else if (flags & (1 << 21)) { // -i
+                printf("unsigned char buf[] = {");
```
Comment (radare) [inline binr/rax2/rax2.c]: space before (

File: binr/rax2/rax2.c [hunk 3]
```diff
@@ -480,7 +482,26 @@ static int rax(char *str, int len, int last) {
...
+                /* resonable amount of bytes per line */
+                int byte_per_col = 12;
+		for (i = 0; i < len-1; i++) {
+                        /* wrapping every N bytes */
+                        if (i % byte_per_col == 0) {
+                                printf("\n  ");
+                        }
+			printf ("0x%02x, ", (ut8) str[i]);
```
Comment (radare) [inline binr/rax2/rax2.c]: tabs

File: binr/rax2/rax2.c [hunk 4]
```diff
@@ -480,7 +482,26 @@ static int rax(char *str, int len, int last) {
...
+		}
+                /* some care for the last element */
+                if (i % byte_per_col == 0) {
+                        printf("\n  ");
```
Comment (radare) [inline binr/rax2/rax2.c]: space before (

File: binr/rax2/rax2.c [hunk 5]
```diff
@@ -480,7 +482,26 @@ static int rax(char *str, int len, int last) {
...
+                }
+                printf ("0x%02x\n", (ut8) str[len-1]);
+		printf ("};\n");
```
Comment (radare) [inline binr/rax2/rax2.c]: tabs

File: binr/rax2/rax2.c [hunk 6]
```diff
@@ -480,7 +482,26 @@ static int rax(char *str, int len, int last) {
...
+                        if (i % byte_per_col == 0) {
+                                printf("\n  ");
+                        }
+			printf ("0x%02x, ", (ut8) str[i]);
+		}
+                /* some care for the last element */
+                if (i % byte_per_col == 0) {
+                        printf("\n  ");
+                }
+                printf ("0x%02x\n", (ut8) str[len-1]);
```
Comment (radare) [inline binr/rax2/rax2.c]: underflow if len == 0

**Comment by PaoloMonti42** [comment]
Thanks for the review. I will fix everything now but there is another problem.
I just realized that the code will not work with 0x00s in the middle because use_stdin stops at null bytes. I can make it stop at EOF, but it will require custom code and the tool invocation will need to change to "rax2 -i file" instead of "rax2 -i < file" is this ok?

File: binr/rax2/rax2.c (hunk 6 above)
Comment (PaoloMonti42) [reply binr/rax2/rax2.c]: The -I case is unreachable with len == 0, should I still check for it?

File: binr/rax2/rax2.c [hunk 7]
```diff
@@ -480,6 +482,26 @@ static int rax(char *str, int len, int last) {
 			free (str);
 		}
 		return false;
+	} else if (flags & (1 << 21)) { // -i
+                printf("%d\n", len);
```
Comment (radare) [inline binr/rax2/rax2.c]: use tabs

Comment (radare) [reply binr/rax2/rax2.c]: missing space before (

File: binr/rax2/rax2.c [hunk 8]
```diff
@@ -480,6 +482,26 @@ static int rax(char *str, int len, int last) {
...
+		printf("unsigned char buf[] = {");
```
Comment (radare) [inline binr/rax2/rax2.c]: missing space before (

Comment (radare) [reply binr/rax2/rax2.c]: maybe add const ?

**Comment by radare** [comment]
Thanks!

**Pull request merged**